### ISSUE TITLE
evaluation

### ISSUE THREAD

**Issue by sun-yue2002** [1e12Leon/RemoteSAM#13]
Hi! I would like to ask whether only the foreground class is considered when calculating the IoU for a sample.

**Comment by sun-yue2002** [comment]
Besides, we noticed that there is a subfile in the JEPGImages file after unzip, is this correct?

<img width="753" height="478" alt="Image" src="https://github.com/user-attachments/assets/f0f420ca-c088-4847-9e88-082c968b7fea" />

**Comment by sun-yue2002** [comment]
> Hi! I would like to ask whether only the foreground class is considered when calculating the IoU for a sample.

Done with this one.

**Comment by 1e12Leon** [comment]
Sorry, I didn't notice earlier. Is there a subfolder? Maybe it was accidentally dragged in during upload... It probably doesn't belong to the training set.

**Comment by sun-yue2002** [comment]
> Sorry, I didn't notice earlier. Is there a subfolder? Maybe it was accidentally dragged in during upload... It probably doesn't belong to the training set.

Yes. There is a subfolder. The content of this subfolder is as follow:

<img width="750" height="873" alt="Image" src="https://github.com/user-attachments/assets/c52a7f7b-f620-446d-ad5a-73e551e1b917" />
It seems thet are the original iSAID dataset. Are these imgs belong to trainig set?

**Comment by sun-yue2002** [comment]
> > Sorry, I didn't notice earlier. Is there a subfolder? Maybe it was accidentally dragged in during upload... It probably doesn't belong to the training set.
> 
> Yes. There is a subfolder. The content of this subfolder is as follow:
> 
> <img alt="Image" width="750" height="873" src="https://private-user-images.githubusercontent.com/127317228/470214101-c52a7f7b-f620-446d-ad5a-73e551e1b917.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNDgzMzAsIm5iZiI6MTc1MzM0ODAzMCwicGF0aCI6Ii8xMjczMTcyMjgvNDcwMjE0MTAxLWM1MmE3ZjdiLWY2MjAtNDQ2ZC1hZDVhLTczZTU1MWUxYjkxNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyNFQwOTA3MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MjUyNjE5YjIwMjUwMzEyZTg0MmZiMjJjNzk5NDllMzhkZGRhNDFiMjAwOTdkNzEyYTk4OTFiMTg2OTM5NDc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pzYmuOybUXtxLNkHUo-MEGNeOkZJhYN3tBPHLzXPnqs"> It seems thet are the original iSAID dataset. Are these imgs belong to trainig set?

<img width="104" height="26" alt="Image" src="https://github.com/user-attachments/assets/5a10173e-f0df-4af2-9d80-767b1e13c3aa" /> 
This is the count num of all imgs in the JPEGimages folder. Is it match with your dataset?

**Comment by 1e12Leon** [comment]
It is indeed the iSAID test set, and it did not conclude in the training. It was probably an oversight on my part to accidentally include it. Thank you for pointing out this issue.

**Comment by 1e12Leon** [comment]
这是我们初步融合其他数据后的图像数量，但有一部分图像最终我们没用在训练里。使用到的图像请以json文件和.p文件中的标注为准。

**Comment by sun-yue2002** [comment]
> It is indeed the iSAID test set, and it did not conclude in the training. It was probably an oversight on my part to accidentally include it. Thank you for pointing out this issue.

Got that. The count num of training data is as follows, so this is the training samples num, right?


<img width="85" height="21" alt="Image" src="https://github.com/user-attachments/assets/c27de679-a74a-4253-be4c-67c1205e5717" />

> 这是我们初步融合其他数据后的图像数量，但有一部分图像最终我们没用在训练里。使用到的图像请以json文件和.p文件中的标注为准。

**Comment by 1e12Leon** [comment]
对的

**Comment by sun-yue2002** [comment]
> 对的

好的！感谢

**Comment by sun-yue2002** [comment]
> 对的

你好！我在处理训练数据集的时候，发现json中的一些训练图像并不在JEPGImages文件夹中，例如下面这张

<img width="1321" height="68" alt="Image" src="https://github.com/user-attachments/assets/232f05e2-90af-4eb2-b098-fc3fa1e900c2" />

所以想请问下解压之后，JEPGImages文件夹中的文件下本身应该有多少影像，我想排除一下文件缺失的原因是否是本地解压问题。

**Comment by 1e12Leon** [comment]
所有的SIOR都没有吗 还是只有这张图

**Comment by 1e12Leon** [comment]
确实有点问题，我最近忙完重新整理一下数据，然后传一个整洁版的数据集过去，抱歉，您这边先自行过滤一下吧，如果没有的图片先跳过一下。

**Comment by sun-yue2002** [comment]
> 所有的SIOR都没有吗 还是只有这张图

不存在的大多是SIOR的图

**Comment by sun-yue2002** [comment]
> 确实有点问题，我最近忙完重新整理一下数据，然后传一个整洁版的数据集过去，抱歉，您这边先自行过滤一下吧，如果没有的图片先跳过一下。

好的，感谢
